src/app/components/abstract/abstract-form-base/readme.md

```diff
@@ -0,0 +1,128 @@
+---
+
+# Abstract Form Base Component
+
+This Angular component serves as a foundational base for managing form controls, validation, and error handling. It provides reusable functionality for managing forms and is designed to be extended by other form components.
+
+## Table of Contents
+
+- [Installation](#installation)
+- [Usage](#usage)
+- [Inputs](#inputs)
+- [Methods](#methods)
+- [Interfaces](#interfaces)
+- [Mocks](#mocks)
+- [Testing](#testing)
+- [Contributing](#contributing)
+
+## Installation
+
+To use the `AbstractFormBaseComponent` in your project, ensure that it is extended by form-related components to handle forms in your Angular application.
+
+```typescript
+import { AbstractFormBaseComponent } from '@components/abstract/abstract-form-base/abstract-form-base.component';
+```
+
+## Usage
+
+This component is designed to be used as a base class for managing forms, validations, and error handling in a reusable and scalable way.
+
+### Example Usage:
+
+```typescript
+import { Component } from '@angular/core';
+import { AbstractFormBaseComponent } from '@components/abstract/abstract-form-base/abstract-form-base.component';
+
+@Component({
+  selector: 'app-my-form',
+  templateUrl: './my-form.component.html',
+})
+export class MyFormComponent extends AbstractFormBaseComponent {
+  // Define specific form controls and methods for your form
+}
+```
+
+## Inputs
+
+`AbstractFormBaseComponent` provides several key inputs that manage form control groups and their validation state.
+
+| Input         | Type       | Description                                         |
+| ------------- | ---------- | --------------------------------------------------- |
+| `formGroup`   | `FormGroup`| The main form group that holds all form controls.    |
+
+## Methods
+
+`AbstractFormBaseComponent` provides utility methods for managing form controls, validating fields, and handling form error messages.
+
+### Common Methods:
+
+- **`initializeForm()`**: Initializes the form with defined controls.
+- **`getControlErrors(controlName: string)`**: Retrieves the error messages for a specific form control.
+- **`isFormValid()`**: Returns `true` if the form is valid, otherwise `false`.
+- **`markAsTouched(controlName: string)`**: Marks the form control as touched for validation purposes.
+- **`displayErrorSummary()`**: Displays summary error messages based on the form’s validation state.
+
+Example:
+
+```typescript
+this.initializeForm();
+const errors = this.getControlErrors('email');
+if (this.isFormValid()) {
+  // Perform actions on valid form submission
+}
+```
```

Comment:
Make sure this section is relevant to the abstract class as some of these methods do not exist in the abstract class